### ISSUE TITLE
Issue in creation of changeset

### ISSUE THREAD

**Issue by bartleboeuf** [aws-cloudformation/rain#213]
with Rain v1.6.0 and also v1.5.0

Since end of last week, we are face to a create change set issue and we don't know what can be the origin. If we use the uploaded file directly in cloudformation console and create a changeset with same deployment role, it's working perfectly.

```
`**/usr/local/bin/rain deploy --no-colour --debug --yes --region cn-north-1 --role-arn $deploy_arn --profile $s3_accountid --s3-bucket cf-templates-m89vz00plzen-cn-north-1 --s3-prefix stack-tools-non-production-main/ --config stack-config.json main.yml stack-tools-non-production-main**

DEBUG: Preparing template 'main.yml'
DEBUG: Loading AWS config
DEBUG: Artifact bucket: cf-templates-m89vz00plzen-cn-north-1
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: transform path: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL
DEBUG: Checking current status of stack 'stack-tools-non-production-main'
DEBUG: Stack exists
DEBUG: Handling parameters
DEBUG:   globalPrefix: <previous value>
DEBUG:   regionPrefix: <previous value>
DEBUG: Creating change set
DEBUG: Template is too large to deploy directly; uploading to S3.
DEBUG: Artifact bucket: cf-templates-m89vz00plzen-cn-north-1
DEBUG: Artifact key: stack-tools-non-production-main/0ac26e813002d829141fb5c8e832b07d1333e7eaba7e55f213f39b2303297cad
panic: error creating changeset: S3 error: The specified bucket does not exist
For more information check http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html [recovered]
    panic: error creating changeset: S3 error: The specified bucket does not exist
For more information check http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

goroutine 1 [running]:
github.com/aws-cloudformation/rain/internal/cmd.execute.func1()
    /__w/rain/rain/internal/cmd/wrap.go:77 +0x15e
panic({0xfbab40?, 0xc0008e78b0?})
    /usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/aws-cloudformation/rain/internal/cmd/deploy.glob..func1(0xc00077ea00?, {0xc0007dc6c0, 0x2, 0x1076146?})
    /__w/rain/rain/internal/cmd/deploy/deploy.go:104 +0x13f3
github.com/spf13/cobra.(*Command).execute(0x28d4b40, {0xc0007dc5a0, 0x11, 0x12})
    /go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x28d3ec0)
    /go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x38d
github.com/spf13/cobra.(*Command).Execute(...)
    /go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/aws-cloudformation/rain/internal/cmd.execute(0xc000102058?)
    /__w/rain/rain/internal/cmd/wrap.go:86 +0x57
github.com/aws-cloudformation/rain/internal/cmd.Execute(0xc0000061a0?)
    /__w/rain/rain/internal/cmd/wrap.go:95 +0x13
main.main()
    /__w/rain/rain/cmd/rain/main.go:23 +0x1a`

```

**Comment by ericzbeard** [comment]
I assume you have confirmed that `cf-templates-m89vz00plzen-cn-north-1` exists. We have another bug that sounds related that I haven't been able to replicate yet in my own account.

**Comment by bartleboeuf** [comment]
We switched to aws cli to enable us to deploy, with the same parameters and the same host. 

```
2023/11/21 14:35:16.968082 Running command /usr/local/bin/aws cloudformation deploy --template-file main.yml --stack-name $stack_name --region cn-north-1 --role-arn $deploy_arn --capabilities "CAPABILITY_NAMED_IAM" "CAPABILITY_IAM" --profile $s3_accountid --s3-bucket cf-templates-m89vz00plzen-cn-north-1 --s3-prefix $stack_name --force-upload --tags $tags

Uploading to stack-tools-non-production-main/0b7e2eff6acbc20d5ceeaf17ca3d1234.template  52022 / 52022.0  (100.00%)

Waiting for changeset to be created..
Waiting for stack create/update to complete
Successfully created/updated stack - stack-tools-non-production-main
```

**Comment by ericzbeard** [comment]
@bartleboeuf Let me know if this is still happening.

**Comment by ericzbeard** [comment]
Oh, I missed that it was in China.

**Comment by ericzbeard** [comment]
Released this fix in 1.7.3

**Comment by kun-shi** [comment]
Hi @ericzbeard if you can support China for the aws-console command, it would be very great.
